### ISSUE TITLE
RightCurlyCheck fails on 7.0 for do while loops

### ISSUE THREAD

**Issue by scottkennedy** [checkstyle/checkstyle#3320]
All our do-while loops are failing the RightCurlyCheck since upgrading to 7.0.

```
do {
    // ...
} while (x > y);
```

```
'}' at column 1 should be alone on a line.

Checks the placement of right curly braces ('}') for else, try, and catch tokens. The policy to verify is specified using the property option.
```

Taking `LITERAL_WHILE` out of the options for `RightCurly` fixes this warning, but then it allows things like:

```
while (x > 5) {
    // ...
} x++;
```

Which is obviously not desired.

The behavior in 6.19 was correct and desired.

**Comment by MEZk** [comment]
@scottkennedy
Please, provide your Checkstyle configuration for RightCurlyCheck.
See http://checkstyle.sourceforge.net/report_issue.html

**Comment by scottkennedy** [comment]
```
<module name="RightCurly"/>
<module name="RightCurly">
    <property name="option" value="alone"/>
    <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
</module>
```

**Comment by MEZk** [comment]
@scottkennedy 
Why do you think that '}' should not be alone on the line? 

```java
do {
    // ...
} while (x > y);
```

You use brace policy 'ALONE' which means that the brace must be alone on the line and not followed by  next part of a multi-block statement, in your case next part of do-while loop is literal 'while'. 

If there were not any violations in previous version, it was a bug.

@romani 
What do you think?

From documentation:

>Checks the placement of right curly braces ('}') for if-else, try-catch-finally blocks, while-loops, for-loops, method definitions, class definitions, constructor definitions, instance and static initialization blocks.

**Comment by scottkennedy** [comment]
I have never seen a code style guide that would suggest putting that `while` anywhere else but right beside the block.

And while I agree it's working "as documented", it's no longer working as desired.

If `DO_WHILE` were a valid token, I would give it the option of "SAME".

**Comment by MEZk** [comment]
@scottkennedy 

Exclude 'LITERAL_DO' from 'tokens'. 

Works OK for me:

````
$ cat TestClass.java
public class TestClass {
    void method() {
        int x = 2;
        int y = 3;
        do {
            int c = 5;
        } while (x > y);
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RightCurly">
    <property name="option" value="alone"/>
    <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT, INSTANCE_INIT"/>
</module>
    </module>
</module>

$ java -jar checkstyle-7.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

Does it help?

**Comment by scottkennedy** [comment]
That worked great, thanks.

**Comment by MEZk** [comment]
@romani 
The problem was due to the fact that we treat LITERAL_DO incorrectly in Checkstyle 6.19. After fix for issue #3090 RightCurlyCheck began validate next part of do-while loop (while-token) correctly.

**Comment by romani** [comment]
@scottkennedy, if your project is open source - please give us a link to it for review.

@MEZk , please create new issue and describe the problem in details (sounds like hotfix release is required).
